### PR TITLE
Rewrite 1_stdlib/Filter.swift test file with StdlibUnittest

### PR DESCRIPTION
**Pull request by frootloops**

### CODE REVIEW

File: test/1_stdlib/Filter.swift [hunk 1]
```diff
@@ -39,59 +43,20 @@ extension LazyFilterCollection where Base : TestProtocol1 {
...
-// Test filtering Collections
-if true {
+FilterTests.test("filtering collections") {
   let f0 = LazyFilterCollection(0..<30) { $0 % 7 == 0 }
-  
-  // CHECK-NEXT: <0, 7, 14, 21, 28>
-  printlnByGenerating(f0)
-  // CHECK-NEXT: <0, 7, 14, 21, 28>
-  printlnByIndexing(f0)
+  expectEqual([0, 7, 14, 21, 28], f0.map { $0 })
```
Comment (gribozavr) [inline test/1_stdlib/Filter.swift]: `expectEqualSequence`, please.

File: test/1_stdlib/Filter.swift [hunk 2]
```diff
@@ -39,59 +43,20 @@ extension LazyFilterCollection where Base : TestProtocol1 {
...
 
-  // Also try when the first element of the underlying sequence
-  // doesn't pass the filter
   let f1 = LazyFilterCollection(1..<30) { $0 % 7 == 0 }
-  
-  // CHECK-NEXT: <7, 14, 21, 28>
-  printlnByGenerating(f1)
-  // CHECK-NEXT: <7, 14, 21, 28>
-  printlnByIndexing(f1)
+  expectEqual([7, 14, 21, 28], f1.map { $0 })
```
Comment (gribozavr) [inline test/1_stdlib/Filter.swift]: `expectEqualSequence`

File: test/1_stdlib/Filter.swift [hunk 3]
```diff
@@ -39,59 +43,20 @@ extension LazyFilterCollection where Base : TestProtocol1 {
...
 
-
-// Test filtering Sequences
-if true {
+FilterTests.test("filtering sequences") {
   let f0 = (0..<30).generate().lazy.filter { $0 % 7 == 0 }
-  
-  // CHECK-NEXT: <0, 7, 14, 21, 28>
-  printlnByGenerating(f0)
+  expectEqual([0, 7, 14, 21, 28], f0.map { $0 })
```
Comment (gribozavr) [inline test/1_stdlib/Filter.swift]: `expectEqualSequence`

File: test/1_stdlib/Filter.swift [hunk 4]
```diff
@@ -39,59 +43,20 @@ extension LazyFilterCollection where Base : TestProtocol1 {
...
 
-  // Also try when the first element of the underlying sequence
-  // doesn't pass the filter
   let f1 = (1..<30).generate().lazy.filter { $0 % 7 == 0 }
-  
-  // CHECK-NEXT: <7, 14, 21, 28>
-  printlnByGenerating(f1)
+  expectEqual([7, 14, 21, 28], f1.map { $0 })
```
Comment (gribozavr) [inline test/1_stdlib/Filter.swift]: `expectEqualSequence`

**Comment by frootloops** [comment]
@gribozavr thanks for comment. done

**Comment by frootloops** [comment]
@gribozavr why do you write ` extension LazyFilterGenerator where Base : TestProtocol1 {...` ?

**Comment by gribozavr** [comment]
@frootloops Yes, the comment explains it:

```swift
 // Check that the generic parameter is called 'Base'.
```

**Comment by gribozavr** [comment]
@frootloops Thank you for working on this, merged!

**Pull request merged**